docs/userguide/entry_point.rst

```diff
@@ -69,20 +73,35 @@ above example, to create a command ``hello-world`` that invokes
         from setuptools import setup
 
         setup(
-            name='timmins',
-            version='0.0.1',
-            packages=['timmins'],
-			# ...
+            # ...,
             entry_points={
-				'console_scripts': [
-					'hello-world=timmins:hello_world',
-				]
-			}
+                'console_scripts': [
+                    'hello-world=timmins:hello_world',
+                ]
+            }
         )
 
+.. tab:: pyproject.toml (**EXPERIMENTAL**) [#experimental]_
+
+   .. code-block:: toml
+
+        [project.scripts]
+        hello-world = "timmins:hello_world"
+
 
 After installing the package, a user may invoke that function by simply calling
-``hello-world`` on the command line.
+``hello-world`` on the command line:
+
+.. code-block:: bash
+
+   $ hello-world
+   Hello world
+
+Note that any function configured as a console script, i.e. ``hello_world()`` in
+this example, should not accept any arguments. If your function requires any input
+from the user, you can use regular command-line argument parsing utilities like
+`argparse <https://docs.python.org/3/library/argparse.html>`_ within the body of
+the function to parse user input.
```

Comment:
```suggestion
the function to parse user input given via :obj:`sys.argv`.
```

Maybe mention `sys.argv` directly so users are aware that they can also access it?